src/compiler/variables.jl

```diff
@@ -87,33 +87,40 @@ function toqubo_variables!(model::VirtualQUBOModel{T}, ::AbstractArchitecture) w
             #       N ≥ log₂(1 + |b - a| / 4τ)
             #
             # where τ is the (absolute) tolerance
-            # TODO: Add τ as parameter
+            # TODO: Add τ as parameter (DONE)
+            # TODO: Move this comment to the documentation
             let
-                τ = MOI.get(model, Tol(), x)
-                e = MOI.get(model, VariableEncoding(), x)
-
-                encode!(e, model, x, a, b, τ)
+                e = MOI.get(model, Attributes.VariableEncodingMethod(), x)
+                n = MOI.get(model, Attributes.VariableEncodingBits(), x)
+
+                if !isnothing(n)
+                    encode!(model, e, x, a, b, n)
+                else
+                    τ = MOI.get(model, Attributes.VariableEncodingATol(), x)
+                    encode!(model, e, x, a, b, τ)
+                end
             end 
         end
     end
 
-    # -*- Discretize Integer Variables 🤔 -*-
+    # Discretize Integer Variables 
     for (x, (a, b)) in ℤ
         if isnothing(a) || isnothing(b)
             error("Unbounded variable $(x) ∈ ℤ")
         else
-            let e = MOI.get(model, VariableEncoding(), x)
-                encode!(e, model, x, a, b)
+            let
+                e = MOI.get(model, Attributes.VariableEncodingMethod(), x)
+                encode!(model, e, x, a, b)
             end
         end
     end
 
-    # -*- Mirror Boolean Variables 😄 -*-
+    # Mirror Boolean Variables 😄 
```

Comment:
I see some emojis left. I'm not the biggest fan of emojis, so please consider removing them.